lib/internal/main/watch_mode.js

```diff
@@ -27,14 +28,29 @@ const { inspect } = require('util');
 const { setTimeout, clearTimeout } = require('timers');
 const { resolve } = require('path');
 const { once } = require('events');
+const { createMatcher } = require('internal/fs/glob');
+const { globSync } = require('fs');
 
 prepareMainThreadExecution(false, false);
 markBootstrapComplete();
 
+function hasGlobPattern(path) {
+  return createMatcher(path).hasMagic();
+}
+
+function handleWatchedPath(path) {
+  if (hasGlobPattern(path)) {
+    const matchedFilesFromGlob = globSync(path);
+    const resolvedMatchedFiles = ArrayPrototypeMap(matchedFilesFromGlob, (path) => resolve(path));
+    return resolvedMatchedFiles;
+  }
+  return resolve(path);
+}
+
 const kKillSignal = convertToValidSignal(getOptionValue('--watch-kill-signal'));
 const kShouldFilterModules = getOptionValue('--watch-path').length === 0;
```

Comment:
```suggestion
const kShouldFilterModules = getOptionValue('--watch-pattern').length === 0;
```
I wonder if this should be a separate flag, and if not if its a breaking change